### PR TITLE
[profile] delete account screen

### PR DESCRIPTION
**Pull request by stephaniewong2**
### Description

[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Built out front end and backend for Delete Account screen. 
### Screenshots

[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review

[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Create a dummy account and navigate to the profile tab. Go to Delete Account and click "Delete Account". This should take you to the Welcome screen. Try to log into the account you just deleted to confirm that we no longer have this account. Clicking "Cancel" should just take you back to the Profile screen.
- Make sure styling matches Figma and works properly on various screen sizes. 
## Next steps
- Remove bottom tab navigation from this screen.
- Add AuthContext

[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources

[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @stephaniewong2

### CODE REVIEW

File: src/app/(BottomTabNavigation)/Profile/DeleteAccount/styles.ts [hunk 1]
```diff
@@ -6,13 +6,64 @@ export default StyleSheet.create({
...
+  },
+  blurb: {
+    fontSize: 16,
+    fontStyle: 'normal',
+    fontWeight: '300',
+    color: colors.black,
+  },
+  buttonContainer: {
+    flexDirection: 'row',
+    marginTop: 365, // im guessing here how do i move the buttons to the bottom of the screen w out just adding a bunch of marginTop for this container
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/Profile/DeleteAccount/styles.ts]: I'd recommend putting all content in a container that spans the whole screen. Then use the "space-between" option of justify-content/align-items (i forget which one).

File: src/app/(BottomTabNavigation)/Profile/DeleteAccount/styles.ts [hunk 2]
```diff
@@ -6,13 +6,64 @@ export default StyleSheet.create({
   container: {
     flex: 1,
     backgroundColor: colors.white,
-    alignItems: 'center',
+    paddingHorizontal: 25,
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/Profile/DeleteAccount/styles.ts]: I'm assuming the padding is for content width? I think it'd be better to have a nested content container within this main container with a percentage width. You could also include the bottom buttons in this container.

File: src/app/(BottomTabNavigation)/Profile/DeleteAccount/index.tsx [hunk 3]
```diff
@@ -11,23 +15,46 @@ import { userInstance, User } from '../../../../types/types';
...
+          style={styles.cancelButton}
+        >
+          <X />
+          <View style={styles.gap} />
+          <Text style={styles.cancelText}>Cancel</Text>
+        </TouchableOpacity>
+        <View style={styles.gap} />
+        <TouchableOpacity onPress={deleteAccount} style={styles.confirmButton}>
+          <WhiteTrash />
+          <View style={styles.gap} />
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/Profile/DeleteAccount/index.tsx]: I'd avoid using empty divs to create spacing. Place both buttons in a div and use "space-between" property.

File: src/app/(BottomTabNavigation)/Profile/DeleteAccount/index.tsx [hunk 4]
```diff
@@ -11,23 +15,46 @@ import { userInstance, User } from '../../../../types/types';
...
   const [currSession, setCurrSession] = useState<User>(userInstance);
+
   useEffect(() => {
     getCurrentUserInfo().then(result => {
       setCurrSession(result);
     });
   }, []);
+
+  const deleteAccount = () => {
+    deleteCurrentUser(currSession.id);
```
Comment (ronniebeggs) [inline src/app/(BottomTabNavigation)/Profile/DeleteAccount/index.tsx:26]: I'm not sure why but my account doesn't get deleted. Weird because you didn't change this function in this PR, right?

**Review by ronniebeggs — CHANGES_REQUESTED** [review]
Looking good so far! Thanks for getting this out so quick!!!!
